src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -4018,8 +4018,10 @@ private ushort getRelocTypeHint(void* target)
                 case TargetArchitecture.X64:
                     return (ushort)RelocType.IMAGE_REL_BASED_REL32;
 
+#if READYTORUN
                 case TargetArchitecture.ARM:
```

Comment:
How much perf does this leave on the table? 

Would it be better to detect and skip the linker generated thunks in the code manager?

Reply:
I don't have the measurements for performance but there are several things to consider:
- The thunks may appear in front of the managed code section or at the end. Either way it can only cover a limited range (+- 16MB) and we would still get a failure if this range is exceeded. Some of the tests are not far away from hitting such limit.
- Detecting the thunks is non-trivial. The exception unwinding table is indexed only by method start address and does not contain the end address or method length. It's assumed that the next method follows the previous one. To detect thunks inserted at the beginning of the managed code section we could look for the lowest address of unwinding info within the section. Detecting thunks at the end of the managed code requires inserting an extra "cantunwind" marker at the end of our managed code to get an idea where the inserted thunks start. The linker doesn't generate any unwinding table information for the thunks.
- Inserting custom label symbols to mark the beginning+end of our managed code wouldn't work when using the multi-object-file mode. Likewise, in that mode the thunks could appear in the middle of the section when multiple object files are merged. That may still be solvable with the "cantunwind" end marker but it's noticably harder to reason about.
- CoreCLR doesn't seem to send the relocation to JIT either. It doesn't ask for relocatable code so the sequence is slightly shorter (2 bytes iirc) but it still doesn't use the short BL as far as I can tell.

Given all the considerations I would prefer to have a correct baseline and optimize later.